>From f3da11a7063ebc4719e5c2b829ac04dc10fb147f Mon Sep 17 00:00:00 2001
From: Ron <ron@debian.org>
Date: Mon, 28 Jul 2014 19:29:45 +0930
Subject: [PATCH] Fix lib location and search path for syslinux >= 5

The changes in b03b133efece84a1f81e41385200ef26e8287cfd were not
sufficient for pxelinux in the netboot tarball to actually boot.
The ldlinux.c32 loader must be located in the tftp root, and the
other new support libraries (libcom32 and libutil) must either be
located there too or have a PATH set to where they can be found.
---
 build/boot/x86/syslinux.cfg | 2 ++
 build/config/x86.cfg        | 3 +++
 2 files changed, 5 insertions(+)

diff --git a/build/boot/x86/syslinux.cfg b/build/boot/x86/syslinux.cfg
index 21dc753..7b2a1ce 100644
--- a/build/boot/x86/syslinux.cfg
+++ b/build/boot/x86/syslinux.cfg
@@ -1,4 +1,6 @@
 # D-I config version 2.0
+# search path for the c32 support libraries (libcom32, libutil etc.)
+path ${SYSDIR}
 include ${SYSDIR}menu.cfg
 default ${SYSDIR}vesamenu.c32
 prompt 0
diff --git a/build/config/x86.cfg b/build/config/x86.cfg
index 456314f..07f99b8 100644
--- a/build/config/x86.cfg
+++ b/build/config/x86.cfg
@@ -391,6 +391,9 @@ arch_netboot_dir: x86_syslinux
 	# dhcp server setups to work without modification.
 	rm -f $(TEMP_NETBOOT_DIR)/pxelinux.0
 	ln -sf $(NETBOOT_PATH)/pxelinux.0 $(TEMP_NETBOOT_DIR)/pxelinux.0
+	# This link is required because pxelinux only looks in the tftp root
+	# for this library (it does a PATH search for any others).
+	ln -sf $(NETBOOT_PATH)/$(BOOT_SCREEN_DIR)/ldlinux.c32 $(TEMP_NETBOOT_DIR)/ldlinux.c32
 
 	set -e; \
 	$(foreach file,$(wildcard $(TEMP_BOOT_SCREENS)/*.txt), \
-- 
2.0.0

